lib/src/sheet/border_style.dart

```diff
@@ -0,0 +1,115 @@
+part of excel;
+
+class Border extends Equatable {
+  late final BorderStyle? borderStyle;
+  late final String? borderColorHex;
+
+  Border({this.borderStyle, String? borderColorHex}) {
+    this.borderColorHex =
+        borderColorHex != null ? _isColorAppropriate(borderColorHex) : null;
+  }
+
+  @override
+  String toString() {
+    return 'Border(borderStyle: $borderStyle, borderColorHex: $borderColorHex)';
+  }
+
+  @override
+  List<Object?> get props => [
+        borderStyle,
+        borderColorHex,
+      ];
+}
+
+class BorderSet extends Equatable {
+  late final Border? leftBorder;
+  late final Border? rightBorder;
+  late final Border? topBorder;
+  late final Border? bottomBorder;
+  late final Border? diagonalBorder;
+  late final bool diagonalBorderUp;
+  late final bool diagonalBorderDown;
+
+  BorderSet({
+    this.leftBorder,
+    this.rightBorder,
+    this.topBorder,
+    this.bottomBorder,
+    this.diagonalBorder,
+    this.diagonalBorderUp = false,
+    this.diagonalBorderDown = false,
+  });
+
+  BorderSet copyWith({
+    Border? leftBorder,
+    Border? rightBorder,
+    Border? topBorder,
+    Border? bottomBorder,
+    Border? diagonalBorder,
+    bool? diagonalBorderUp,
+    bool? diagonalBorderDown,
+  }) {
+    return BorderSet(
+      leftBorder: leftBorder ?? this.leftBorder,
+      rightBorder: rightBorder ?? this.rightBorder,
+      topBorder: topBorder ?? this.topBorder,
+      bottomBorder: bottomBorder ?? this.bottomBorder,
+      diagonalBorder: diagonalBorder ?? this.diagonalBorder,
+      diagonalBorderUp: diagonalBorderUp ?? this.diagonalBorderUp,
+      diagonalBorderDown: diagonalBorderDown ?? this.diagonalBorderDown,
+    );
+  }
+
+  @override
+  List<Object?> get props => [
+        leftBorder,
+        rightBorder,
+        topBorder,
+        bottomBorder,
+        diagonalBorder,
+        diagonalBorderUp,
+        diagonalBorderDown,
+      ];
+}
+
+enum BorderStyle {
+  None,
+  DashDot,
+  DashDotDot,
+  Dashed,
+  Dotted,
+  Double,
+  Hair,
+  Medium,
+  MediumDashDot,
+  MediumDashDotDot,
+  MediumDashed,
+  SlantDashDot,
+  Thick,
+  Thin,
+}
+
+Map<BorderStyle, String> _borderStyleStringMap = {};
+
+Map<BorderStyle, String> _getBorderStyleStringMap() {
+  if (_borderStyleStringMap.isEmpty) {
+    for (BorderStyle borderStyle in BorderStyle.values) {
+      var borderStyleStr =
+          borderStyle.toString().replaceAll('BorderStyle.', '');
+      borderStyleStr =
+          borderStyleStr[0].toLowerCase() + borderStyleStr.substring(1);
+      _borderStyleStringMap[borderStyle] = borderStyleStr;
+    }
+  }
+
+  return _borderStyleStringMap;
+}
+
+BorderStyle? getBorderStyleByName(String name) {
```

Comment:
Suggestion:

```dart
const _borderStyleMapped = <String, BorderStyle>{
  'none': BorderStyle.None,
  'dashDot': BorderStyle.DashDot,
  'dashDotDot': BorderStyle.DashDotDot,
  'dashed': BorderStyle.Dashed,
  'dotted': BorderStyle.Dotted,
  'double': BorderStyle.Double,
  'hair': BorderStyle.Hair,
  'medium': BorderStyle.Medium,
  'mediumDashDot': BorderStyle.MediumDashDot,
  'mediumDashDotDot': BorderStyle.MediumDashDotDot,
  'mediumDashed': BorderStyle.MediumDashed,
  'slantDashDot': BorderStyle.SlantDashDot,
  'thick': BorderStyle.Thick,
  'thin': BorderStyle.Thin,
};

BorderStyle? getBorderStyleByName(String name) => _borderStyleMapped[name];
```
Removes iteration.

What's your thought ?

Reply:
To be honest, I don't really like to repeat myself in code (DRY: Don't Repeat Yourself)... So having an hardcoded map that has the same information than the enum is not something I enjoy much.

That was the reason why I proposed to create the `_borderStyleStringMap` programmatically. But I agree this looks heavy code.

Your previous proposal (to redefine the enum with the associated strings) let me think the problem differently :
```
BorderStyle? getBorderStyleByName(String name) =>
    BorderStyle.values.firstWhereOrNull((e) =>
        e.toString().toLowerCase() == 'borderstyle.' + name.toLowerCase());
```

This way, no intermediate map (neither hardcoded, nor pre-generated) is needed.

As the project owner, please let me know your preference and I'll put it in the code.

Reply:
I implemented my DRY proposal.